File: app/src/main/java/com/jakebarnby/filemanager/models/SourceListener.java
package com.jakebarnby.filemanager.models;

import com.jakebarnby.filemanager.models.files.SourceFile;
import com.jakebarnby.filemanager.util.TreeNode;

public interface SourceListener {
    void onCheckPermissions(String permission, int requestCode);
    void onNoConnection();
    void onLoadStarted();
    void onLoadAborted();
    void onAuthenticationComplete(boolean success);
    void onLoadComplete(TreeNode<SourceFile> rootFile);
}

File: app/src/main/java/com/jakebarnby/filemanager/models/SourceListener.kt
package com.jakebarnby.filemanager.models

interface SourceListener {
    fun onCheckForToken()
    fun onCheckPermissions(name: String, requestCode: Int)
    fun onConnect()
    fun onNoConnection()
    fun onLoadStarted()
    fun onLoadAborted()
    fun onLoadError(errorMessage: String?)
    fun onLoadComplete(rootFile: SourceFile)
    fun onLogout()
}